### PR TITLE
APIDIFF: Fixing bugs in content generation

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR adds necessary updates so that the content can be generated correctly for libraries, such as azcore, azidentity, armcore, etc. It has been tested with the previously mentioned modules.

### CODE REVIEW

File: tools/apidiff/exports/exports.go [hunk 1]
```diff
@@ -110,10 +118,15 @@ func (c *Content) addConst(pkg Package, g *ast.GenDecl) {
 				// const FooConst = FooType("value")
 				co.Type = pkg.getText(ce.Fun.Pos(), ce.Fun.End())
 				v = pkg.getText(ce.Args[0].Pos(), ce.Args[0].End())
+			} else if ce, ok := vs.Values[0].(*ast.BinaryExpr); ok {
+				// const FooConst = "value" + Bar
+				co.Type = "*ast.BinaryExpr"
```
Comment (jhendrixMSFT) [inline tools/apidiff/exports/exports.go:123]: Why do we need this?

File: tools/apidiff/exports/exports.go [hunk 2]
```diff
@@ -110,10 +118,15 @@ func (c *Content) addConst(pkg Package, g *ast.GenDecl) {
...
+				v = pkg.getText(ce.X.Pos(), ce.Y.End())
 			} else {
-				panic(fmt.Sprintf("unhandled case for adding constant: %s", pkg.getText(vs.Pos(), vs.End())))
+				panic("unhandled case for adding constant")
```
Comment (jhendrixMSFT) [inline tools/apidiff/exports/exports.go:126]: Can we keep the text for the line in the panic message?

File: tools/apidiff/exports/exports.go (hunk 1 above)
Comment (catalinaperalta) [reply tools/apidiff/exports/exports.go:123]: If I recall correctly, the Type field can't be empty so I put this arbitrary value in it. I could put the content of the const in it instead if you think that's better?

Comment (jhendrixMSFT) [reply tools/apidiff/exports/exports.go:123]: Oh yes I remember now.  It's used during comparison to see if the `const` type has changed.  I think what you have is fine.

File: tools/apidiff/exports/exports.go (hunk 2 above)
Comment (catalinaperalta) [reply tools/apidiff/exports/exports.go]: Done

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**